docs/react/week-3/learning-objectives.md

```diff
@@ -6,8 +6,14 @@ sidebar_label: Learning Objectives
 
 ## React Week 3
 
-### Data fetching (advanced)
+### Data fetching
 
+Show some data from an API in a component.
+
+- Can explain why a loading state is necessary when fetching data as the result is not available on the initial render
+- Be able to trigger `fetch()` using the `useEffect()` callback, with an empty array (`[]`) as the 2nd dependencies argument
+- Be able to store data from a successful API call in state
+- Be able to catch an unsuccessful API call, store the error in state and render an error message
```

Comment:
We don't cover this in-class - is this expected to be pure carry-over from having done `fetch` before? Worth throwing in an exercise, maybe with an API we publish?

Reply:
> is this expected to be pure carry-over from having done fetch before?

Hmm, good point. I think the concept of an error state would be new.

I am a bit concerned about time in week 3 though.

Reply:
I'd suggest just dropping the objective. We kind of hand-wave over `fetch` errors in JS2 - if we're going to try to teach handling them better in React, we should start by bulking up their treatment in JS2.